src/activities/Elsa.Activities.BlobStorage/Activities/ReadBlob.cs

```diff
@@ -0,0 +1,46 @@
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Elsa.ActivityResults;
+using Elsa.Attributes;
+using Elsa.Services;
+using Elsa.Services.Models;
+using Storage.Net.Blobs;
+
+// ReSharper disable once CheckNamespace
+namespace Elsa.Activities.BlobStorage
+{
+    [Action(
+        Category = "BlobStorage",
+        Description = "Reads a blob from the storage engine",
+        Outcomes = new[] { OutcomeNames.Done }
+    )]
+    public class ReadBlob : Activity
+    {
+        public ReadBlob(IBlobStorage storage)
+        {
+            _storage = storage;
+        }
+        private readonly IBlobStorage _storage;
+
+        [ActivityProperty(Hint = "The ID assigned to the blob.")]
+        [Required]
+        public string BlobID { get; set; }
+        [ActivityProperty(Hint = "The bytes")]
+        public byte[] Bytes { get; set; }
+        [ActivityProperty(Hint = "The output file path, alternative to Bytes")]
+        public string FilePath { get; set; }
+
+        protected override async ValueTask<IActivityExecutionResult> OnExecuteAsync(ActivityExecutionContext context)
+        {
+            if (string.IsNullOrWhiteSpace(BlobID))
+                throw new System.Exception($"BlobID must have a value");
+            if (!string.IsNullOrWhiteSpace(FilePath))
+                await _storage.ReadToFileAsync(BlobID, FilePath, context.CancellationToken);
+            else
+                Bytes = await _storage.ReadBytesAsync(BlobID);
```

Comment:
Setting `Bytes` like this won't do much.  What you might want to do instead is return the bytes as output, e.g.:

```csharp
var bytes = await _storage.ReadBytesAsync(BlobId);
return Done(bytes);
```

Then the blob bytes will be available to the workflow as this activity's output.

Reply:
@sfmskywalker 
Thank for your feedback. Just a question here: from DB serialization perspective, it is safe to pass a byte[] as Activity result?
I mean, Bytes should be available for the next activity to consume, but it shouldn't be persisted in the DB for the following reasons:
-) low serialization performance
-) big db records/documents
-) conflict in serialization mappings
Ref: https://elsa-workflows.github.io/elsa-core/docs/extensibility-custom-activities#state

Reply:
Good point. Right now the user can configure whether a given activity's output should be persisted or not, so perhaps that's enough. But perhaps we need to consider allowing the activity itself to control whether or not the output gets persisted or not.

The simplest solution for now would be to:

1. Return the bytes as output.
2. Update Elsa to *not* store output by default, but instead require the user to explicitly configure an activity's "Persist Output" setting.